File: test/plugin/windows-cpp/TestOneWayStruct.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for details.

#include "pch.h"
#include "TestOneWayStruct.h"

using namespace Microsoft::VisualStudio::C3P::Test;
using namespace Platform;

TestOneWayStruct::TestOneWayStruct(Platform::String^ initialValue)
{
    this->value = initialValue;
}

Platform::String^ TestOneWayStruct::Value::get()
{
    return this->value;
}

File: test/plugin/windows-cs/TestOneWayStruct.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for details.

namespace Microsoft.C3P.Test
{
    public sealed class TestOneWayStruct
    {
        internal TestOneWayStruct(string initialValue)
        {
            this.Value = initialValue;
        }

        public string Value { get; private set; }
    }
}
